test/test_nn.py

```diff
@@ -4237,6 +4237,10 @@ def test_CTCLoss_cudnn(self):
             res2 = torch.nn.functional.ctc_loss(log_probs, targets.cuda().long(), input_lengths, target_lengths)
         self.assertEqual(res, expected)
         self.assertEqual(res2, res)
+        with self.assertRaises(RuntimeError):
+            input_lengths = torch.tensor(input_lengths).type('torch.FloatTensor')
```

Comment:
- I think we might move these to their own `test_...` method that isn't `cudnn`.
- Can you use `.float()` or `.to(dtype=torch.float)` instead, please? I think `.type` is a bit legacyish.
- Also I'd probably do the two float-tensor things separately.
- We would also want a test for the maximum size checks. Could you please add that?
  For those, we do want to check cpu and gpu, you could look in the other tests for a pattern for how to check both.

Reply:
Working on those right now